golem-worker-service-base/src/gateway_api_definition/http/api_oas_convert.rs

```diff
@@ -0,0 +1,1097 @@
+    let mut binding_info = serde_json::Map::new();
+
+    match route.binding.binding_type {
+        Some(GatewayBindingType::Default) => {
+            binding_info.insert(
+                "binding-type".to_string(),
+                serde_json::Value::String("default".to_string()),
+            );
+            add_common_binding_info(&mut binding_info, route);
+        }
+        Some(GatewayBindingType::HttpHandler) => {
+            binding_info.insert(
+                "binding-type".to_string(),
+                serde_json::Value::String("http-handler".to_string()),
+            );
+            add_common_binding_info(&mut binding_info, route);
+        }
+        Some(GatewayBindingType::FileServer) => {
+            binding_info.insert(
+                "binding-type".to_string(),
+                serde_json::Value::String("file-server".to_string()),
+            );
+            add_common_binding_info(&mut binding_info, route);
+        }
+        Some(GatewayBindingType::CorsPreflight) => {
+            add_cors_preflight_binding_info(&mut binding_info, route);
+        }
+        Some(GatewayBindingType::SwaggerUi) => {
+            binding_info.insert(
+                "binding-type".to_string(),
+                serde_json::Value::String("swagger-ui".to_string()),
+            );
+        }
+        None => {
+            binding_info.insert(
+                "binding-type".to_string(),
+                serde_json::Value::String("default".to_string()),
+            );
+        }
+    }
+
+    binding_info
+}
+
+// Helper function: Adds common binding info
+fn add_common_binding_info(
+    binding_info: &mut serde_json::Map<String, serde_json::Value>,
+    route: &RouteResponseData,
+) {
+    if let Some(worker_name) = &route.binding.worker_name {
+        binding_info.insert(
+            "worker-name".to_string(),
+            serde_json::Value::String(worker_name.clone()),
+        );
+    }
+
+    if let Some(versioned_component) = &route.binding.component {
+        binding_info.insert(
+            "component-name".to_string(),
+            serde_json::Value::String(versioned_component.name().to_string()),
+        );
+        binding_info.insert(
+            "component-version".to_string(),
+            serde_json::Value::Number(serde_json::Number::from(versioned_component.version())),
+        );
+    }
+
+    if let Some(key) = &route.binding.idempotency_key {
+        binding_info.insert(
+            "idempotency-key".to_string(),
+            serde_json::Value::String(key.clone()),
+        );
+    }
+
+    if let Some(response) = &route.binding.response {
+        binding_info.insert(
+            "response".to_string(),
+            serde_json::Value::String(response.clone()),
+        );
+    }
+}
+
+// Helper function: Adds cors-preflight binding info
+// Converts the cors-preflight binding info to a response string
+fn add_cors_preflight_binding_info(
```

Comment:
A binding is per route.  Per route, there cannot exist multiple binding info
This is more tangible if you use CompiledRoute instead of the `RouteResponseData` which is flattened structure making all binding optional